Bug in update_fractional_ion

```
        atm_index = 0
        for atom in molecule:
>           charge = (1 - fraction) * initial_force[atm_index]["charge"] + fraction * final_force["charge"]
E           TypeError: list indices must be integers or slices, not str /home/rustenburg/miniconda3/lib/python3.6/site-packages/saltswap/swapper.py:811 

```

My guess is that it should be `final_force[atm_index]["charge"]` instead of `final_force["charge"]`. Will put in a PR with fix in a minute, and a unit test for this function to capture any other errors with this feature.

--- a/saltswap/swapper.py
+++ b/saltswap/swapper.py
@@ -808,9 +808,9 @@ class Swapper(object):
         molecule = [atom for atom in self.mutable_residues[residue_index].atoms()]
         atm_index = 0
         for atom in molecule:
-            charge = (1 - fraction) * initial_force[atm_index]["charge"] + fraction * final_force["charge"]
-            sigma = (1 - fraction) * initial_force[atm_index]["sigma"] + fraction * final_force["sigma"]
-            epsilon = (1 - fraction) * initial_force[atm_index]["epsilon"] + fraction * final_force["epsilon"]
+            charge = (1 - fraction) * initial_force[atm_index]["charge"] + fraction * final_force[atm_index]["charge"]
+            sigma = (1 - fraction) * initial_force[atm_index]["sigma"] + fraction * final_force[atm_index]["sigma"]
+            epsilon = (1 - fraction) * initial_force[atm_index]["epsilon"] + fraction * final_force[atm_index]["epsilon"]
             self.forces_to_update.setParticleParameters(atom.index, charge=charge, sigma=sigma, epsilon=epsilon)
             atm_index += 1
 

--- a/saltswap/tests/test_simulation.py
+++ b/saltswap/tests/test_simulation.py
@@ -372,3 +372,21 @@ class TestWaterBoxSimulation(object):
                                             delta_chem=0, nprop=1, npert=5)
         sampler.multimove(nmoves=2, mdsteps=1, saltsteps=1)
 
+    def test_update_fractional_ion(self):
+        """
+        Test the functionality of the Swapper.update_fractional_ion method.
+        """
+        integrator,context,swapper = self._create_langevin_system()
+        #Randomly select a water molecule:
+        water_index = np.random.choice(a=np.where(swapper.stateVector == 0)[0], size=1)
+
+        #Completely change this molecule's non-bonded parameters to an anion:
+        swapper.update_fractional_ion(water_index, swapper.water_parameters, swapper.anion_parameters, fraction=1.0)
+
+        #Push change to context:
+        swapper.forces_to_update.updateParametersInContext(context)
+
+        # Update the state vector to keep track of this change: 0 is for water, 1 is for cations, and 2 is for anions.
+        swapper.stateVector[water_index] = 2
+
+

--- a/saltswap/tests/test_simulation.py
+++ b/saltswap/tests/test_simulation.py
@@ -378,7 +378,7 @@ class TestWaterBoxSimulation(object):
         """
         integrator,context,swapper = self._create_langevin_system()
         #Randomly select a water molecule:
-        water_index = np.random.choice(a=np.where(swapper.stateVector == 0)[0], size=1)
+        water_index = np.random.choice(a=np.where(swapper.stateVector == 0)[0], size=1)[0]
 
         #Completely change this molecule's non-bonded parameters to an anion:
         swapper.update_fractional_ion(water_index, swapper.water_parameters, swapper.anion_parameters, fraction=1.0)

--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@ setup(
     name='saltswap',
     author='Gregory A. Ross',
     author_email='gregory.ross@choderalab.org',
-    version='0.1dev',
+    version='0.5.2',
     url='https://github.com/choderalab/saltswap',
     packages=['saltswap', 'saltswap.tests'],
     license='MIT',

